### ISSUE TITLE
lein uberjar fails with duplicate entry: leiningen/protobuf.clj

### ISSUE THREAD

**Issue by aphyr** [ninjudd/clojure-protobuf#18]
Ever seen this before? I'm wondering if it's a conflict with the lein plugin somehow... digging into the source.

```
aphyr@azimuth:~/ustate/reimann master$ lein uberjar
Compiling reimann.bin
Compiling reimann.bin
Compilation succeeded.
Created /home/aphyr/ustate/reimann/reimann-0.0.1-SNAPSHOT.jar
Including reimann-0.0.1-SNAPSHOT.jar
Including tools.macro-0.1.1.jar
Including clojure-1.2.0.jar
Including lamina-0.4.0.jar
Including commons-compress-1.3.jar
Including httpclient-4.0.3.jar
Including gloss-0.2.0.jar
Including netty-3.2.5.Final.jar
Including commons-codec-1.4.jar
Including potemkin-0.1.0.jar
Including commons-io-1.4.jar
Including useful-0.7.4-alpha4.jar
Including clojure-contrib-1.2.0.jar
Including protobuf-0.6.0-beta4.jar
Exception in thread "main" java.lang.RuntimeException: java.util.zip.ZipException: duplicate entry: leiningen/protobuf.clj (NO_SOURCE_FILE:0)
    at clojure.lang.Compiler.eval(Compiler.java:5440)
    at clojure.lang.Compiler.eval(Compiler.java:5391)
    at clojure.core$eval.invoke(core.clj:2382)
    at clojure.main$eval_opt.invoke(main.clj:235)
    at clojure.main$initialize.invoke(main.clj:254)
    at clojure.main$script_opt.invoke(main.clj:270)
    at clojure.main$main.doInvoke(main.clj:354)
    at clojure.lang.RestFn.invoke(RestFn.java:457)
    at clojure.lang.Var.invoke(Var.java:377)
    at clojure.lang.AFn.applyToHelper(AFn.java:172)
    at clojure.lang.Var.applyTo(Var.java:482)
    at clojure.main.main(main.java:37)
Caused by: java.lang.RuntimeException: java.util.zip.ZipException: duplicate entry: leiningen/protobuf.clj
    at clojure.lang.LazySeq.sval(LazySeq.java:47)
    at clojure.lang.LazySeq.seq(LazySeq.java:56)
    at clojure.lang.Cons.next(Cons.java:39)
    at clojure.lang.RT.next(RT.java:560)
    at clojure.core$next.invoke(core.clj:61)
    at clojure.core$r.invoke(core.clj:799)
    at clojure.core$into.invoke(core.clj:2614)
    at leiningen.uberjar$include_dep.invoke(uberjar.clj:47)
    at clojure.lang.AFn.applyToHelper(AFn.java:172)
    at clojure.lang.AFn.applyTo(AFn.java:151)
    at clojure.core$apply.invoke(core.clj:544)
    at clojure.core$partial$fn__3684.doInvoke(core.clj:2011)
    at clojure.lang.RestFn.invoke(RestFn.java:421)
    at clojure.core$r.invoke(core.clj:799)
    at leiningen.uberjar$write_components.invoke(uberjar.clj:54)
    at leiningen.uberjar$uberjar$fn__1571.invoke(uberjar.clj:93)
    at leiningen.uberjar$uberjar.invoke(uberjar.clj:86)
    at leiningen.uberjar$uberjar.invoke(uberjar.clj:97)
    at clojure.lang.Var.invoke(Var.java:365)
    at clojure.lang.AFn.applyToHelper(AFn.java:161)
    at clojure.lang.Var.applyTo(Var.java:482)
    at clojure.core$apply.invoke(core.clj:542)
    at leiningen.core$apply_task.invoke(core.clj:264)
    at leiningen.core$_main.doInvoke(core.clj:331)
    at clojure.lang.RestFn.invoke(RestFn.java:410)
    at clojure.lang.AFn.applyToHelper(AFn.java:161)
    at clojure.lang.RestFn.applyTo(RestFn.java:132)
    at clojure.core$apply.invoke(core.clj:542)
    at leiningen.core$_main.invoke(core.clj:334)
    at user$eval361.invoke(NO_SOURCE_FILE:1)
    at clojure.lang.Compiler.eval(Compiler.java:5424)
    ... 11 more
Caused by: java.util.zip.ZipException: duplicate entry: leiningen/protobuf.clj
    at java.util.zip.ZipOutputStream.putNextEntry(ZipOutputStream.java:192)
    at sun.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:90)
    at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28)
    at leiningen.uberjar$copy_entries$iter__1546__1550$fn__1551.invoke(uberjar.clj:36)
    at clojure.lang.LazySeq.sval(LazySeq.java:42)
    ... 41 more
```

**Comment by Raynes** [comment]
You need to get Leiningen from git and checkout the 1.x branch. This will be the case until Leiningen 1.7.0 is released. There was a bug regarding having both Clojure and Java source code in the same source path. If that fixes your problem, please let me know so I can close this issue.

**Comment by Raynes** [comment]
I'm going to go ahead and close this, since I've got several people using it against the 1.x branch that don't have this problem. If you're still having the problem and you've definitely got the 1.x branch, pop it back open.

**Comment by aphyr** [comment]
Yeah, this was against 1.x cloned on tuesday. traveling now but will try and confirm an sha1 Monday.

**Comment by Raynes** [comment]
Alright, I'll reopen until we've got this figured out then.

**Comment by aphyr** [comment]
Yep, I can confirm this happens on 1.x.

``` bash
aphyr@azimuth:~/leiningen 1.x$ git status
# On branch 1.x
nothing to commit (working directory clean)
aphyr@azimuth:~/leiningen 1.x$ git rev-parse HEAD
4e098f2889b5fb660c7a9532b94dc63e75f96833
```

Project.clj:

``` clojure
(defproject reimann "0.0.1-SNAPSHOT"
  :description "reimann: folds events into states"
  :dependencies [
    [clojure "1.2.0"]
    [aleph "0.2.0"]
    [protobuf "0.6.0-beta4"]
  ]
  :dev-dependencies [
    [protobuf "0.6.0-beta4"]
  ]
  :aot [reimann.bin]
  :main reimann.bin
  :disable-implicit-clean true
)
```

**Comment by aphyr** [comment]
Fixed after protobuf 0.6.0-beta5, lein 1.x, rm -rf lib/, full rebuild of project.
